src/contexts/NotificationsProvider.tsx

```
@@ -0,0 +1,116 @@
+  unsubscribe(notificationType: NotificationTypes): Promise<void>;
+}>({
+  subscriptions: {
+    [BalanceNotificationTypes.BALANCE_CHANGES]: false,
+    [NewsNotificationTypes.PRODUCT_ANNOUNCEMENTS]: false,
+    [NewsNotificationTypes.OFFERS_AND_PROMOTIONS]: false,
+    [NewsNotificationTypes.MARKET_NEWS]: false,
+    [NewsNotificationTypes.PRICE_ALERTS]: false,
+  },
+  async syncSubscriptions() {},
+  async subscribe() {},
+  async unsubscribe() {},
+});
+
+export function NotificationsContextProvider({ children }: { children: any }) {
+  const { request, events } = useConnectionContext();
+
+  const [subscriptions, setSubscriptions] = useState<
+    Record<NotificationTypes, boolean>
+  >({
+    [BalanceNotificationTypes.BALANCE_CHANGES]: true,
+    [NewsNotificationTypes.PRODUCT_ANNOUNCEMENTS]: true,
+    [NewsNotificationTypes.OFFERS_AND_PROMOTIONS]: true,
+    [NewsNotificationTypes.MARKET_NEWS]: true,
+    [NewsNotificationTypes.PRICE_ALERTS]: true,
+  });
+
+  const syncSubscriptions = useCallback(async () => {
+    const result = await request<GetNotificationSubscriptions>({
+      method: ExtensionRequest.NOTIFICATION_GET_SUBSCRIPTIONS,
+      params: [],
+    });
+
+    setSubscriptions(result);
+  }, [request]);
+
+  const subscribe = useCallback(
+    async (notificationType: NotificationTypes) => {
+      await request<SubscribeToNotification>({
+        method: ExtensionRequest.NOTIFICATION_SUBSCRIBE,
+        params: notificationType,
+      }).finally(() => syncSubscriptions());
+    },
+    [request, syncSubscriptions],
+  );
+
+  const unsubscribe = useCallback(
+    async (notificationType: NotificationTypes) => {
+      await request<UnsubscribeFromNotification>({
+        method: ExtensionRequest.NOTIFICATION_UNSUBSCRIBE,
+        params: notificationType,
+      }).finally(() => syncSubscriptions());
+    },
+    [request, syncSubscriptions],
+  );
+
+  useEffect(() => {
+    syncSubscriptions().catch(
+      // noop
+      () => {},
```

I think instead of swallowing every error here, we should check if the extension is unlocked first (either here, or in the `GetNotificationSubscriptions` handler).

Then instead of a `noop`, we could actually log the error here so we know if something ever goes wrong.

---

here: [58de4c5](https://github.com/ava-labs/core-extension/pull/237/commits/58de4c5aa8be047710311283f376ed0f04e1ee8d)